plugins/synaptics-rmi/fu-synaptics-rmi-firmware.c

```diff
@@ -574,6 +640,40 @@ fu_synaptics_rmi_firmware_get_sig_size(FuSynapticsRmiFirmware *self)
 	return self->sig_size;
 }
 
+GPtrArray *
+fu_synaptics_rmi_firmware_get_image_by_id(FuFirmware *firmware, const gchar *id)
+{
+	g_autoptr(GPtrArray) imgs = NULL;
+	g_autoptr(GBytes) img = NULL;
+	g_autoptr(GBytes) img_sig = NULL;
+	g_autoptr(GError) error_local = NULL;
+	imgs = g_ptr_array_new();
+
+	img = fu_firmware_get_image_by_id_bytes(firmware, id, &error_local);
```

Comment:
you need to handle the error

Reply:
I'd be much happier if fu_synaptics_rmi_firmware_get_image_by_id() had a GError parameter...